drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/rightpanel/ListGroupItemViewImpl.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.drools.workbench.screens.scenariosimulation.client.rightpanel;
+
+import javax.enterprise.context.Dependent;
+
+import com.google.gwt.dom.client.DivElement;
+import com.google.gwt.dom.client.Document;
+import com.google.gwt.dom.client.LIElement;
+import com.google.gwt.dom.client.SpanElement;
+import com.google.gwt.dom.client.UListElement;
+import com.google.gwt.event.dom.client.ClickEvent;
+import com.google.gwt.user.client.ui.Widget;
+import org.jboss.errai.ui.shared.api.annotations.DataField;
+import org.jboss.errai.ui.shared.api.annotations.EventHandler;
+import org.jboss.errai.ui.shared.api.annotations.Templated;
+
+@Dependent
+@Templated
+public class ListGroupItemViewImpl implements ListGroupItemView {
+
+    public static final String FA_ANGLE_DOWN = "fa-angle-down";
+    public static final String LIST_VIEW_PF_EXPAND_ACTIVE = "list-view-pf-expand-active";
+    public static final String HIDDEN = "hidden";
+
+    @DataField("listGroupItem")
+    DivElement listGroupItem = Document.get().createDivElement();
+
+    @DataField("listGroupItemHeader")
+    DivElement listGroupItemHeader = Document.get().createDivElement();
+
+    @DataField("listGroupItemContainer")
+    DivElement listGroupItemContainer = Document.get().createDivElement();
+
+    @DataField("faAngleRight")
+    SpanElement faAngleRight = Document.get().createSpanElement();
+
+    @DataField("fullClassName")
+    DivElement fullClassName = Document.get().createDivElement();
+
+    @DataField("factProperties")
+    UListElement factProperties = Document.get().createULElement();
+
+    Presenter presenter;
+
+    String factName;
+
+    String factType;
```

Comment:
It is a bit dangerous to have any other fields than UI components and the presenter. This leaks the model into the view and will likely have bugs due to out of sync issues with Model, view and view fields. The data is already set into the widgets, if you need to store the fields keep them in the Presenter or in the Model

Reply:
@Rikkola As we talked about, for that specific case probably it makes sense to keep those tow properties. In general, I agree to have a separated model